app/controllers/session_shares_controller.rb

```diff
@@ -0,0 +1,20 @@
+class SessionSharesController < ApplicationController
+  before_action :set_session, only: %i[ create ]
+  before_action :authenticate_user!
```

Comment:
I noticed that you have this method call in all of your controllers. You can remove all of this boilerplate by moving it to the `ApplicationController` that all of the other controllers inherit from.

Once you have moved this `before_action` callback there, you can remove it from all of the other ones.

If you have a random controller that doesn't like that (like a user session controller responsible for logging in/out), then you can add a `skip_before_action :authenticate_user!` to just that controller.